rpmio/rpmlog.c

```diff
@@ -237,6 +237,89 @@ const char * rpmlogLevelPrefix(rpmlogLvl pri)
     return prefix;
 }
 
+static const char * rpmlogLevelColor(rpmlogLvl pri)
+{
+    return rpmlogMsgPrefixColor[pri&0x7];
+}
+
+static int rpmlogDefault(FILE *stdlog, rpmlogRec rec)
+{
+    static const char fubar[] =
+	"Error occurred during writing of a log message";
+    FILE *msgout = (stdlog ? stdlog : stderr);
+    const char * colorOn = isatty(fileno(msgout))
```

Comment:
I would prefer to have some macro/option to force enabling/disabling colors (like a --color[=WHEN] where WHEN is "auto", "always", "never")...

because when you would be passing that to `less`, you could pass `-R` and it will show you colors in pager... Quite useful.